staging/src/k8s.io/component-base/metrics/counter.go

```diff
@@ -295,12 +299,51 @@ type CounterVecWithContext struct {
 	ctx context.Context
 }
 
+// CounterVecWithContextWithCounter is the wrapper of CounterVecWithContext with counter.
+type CounterVecWithContextWithCounter struct {
+	*CounterVecWithContext
+	counter CounterMetric
+}
+
+// withExemplar attaches an exemplar to the metric.
+func (vcc *CounterVecWithContextWithCounter) withExemplar(v float64) {
```

Comment:
why do we have the notion of Exemplar at a vector level?

normally users would add the exemplar to the right counter themselves by running:
```
counterVec.WithLabels("my_label").AddWithExemplar()
```
because WithLabels returns a Counter object we shouldn't need to add a notion of exemplar to CounterVec

Reply:
Writing the patch kept bugging me because Prometheus itself has no exemplar methods on vector receivers and we would essentially be deviating, which is why decided to leave it out earlier for counter vectors, and decided to go with this now. But the suggestion definitely makes sense to me and would be the right pattern to adhere to.

Reply:
Actually, now that I think about it, I remember bringing this up during the design discussion and we decided that doing `AddWithExemplars` implicitly would allow for no span and trace ID "misses" in cases where contextual counters are incremented using `Add/Inc` instead of `AddWithExemplars`, albeit this is different from what Prometheus does currently (an explicit `AddWithExemplars` method).

cc @logicalhan @richabanker for visibility.

Reply:
I believe we need to make the choice between convenience (implicit `AddWithExemplars`) and convention here.

Reply:
ACK, `.AddWithExemplar()` has been implicitly incorporated into `.Add()` up until this point, so we could choose to adhere to the same pattern for the future, to keep a consistent `component-base` API across vectors and scalars.